So without the default sort key the Polyhedron test (`py.test -k test_args`) fails since it tries to sort a bunch of Relationals which are not naturally orderable. The old version of this function got around this by using the `default_sort_key` method.

```
________________ test_sympy__combinatorics__polyhedron__Polyhedron _________________

    def test_sympy__combinatorics__polyhedron__Polyhedron():
        from sympy.combinatorics.permutations import Permutation
        from sympy.combinatorics.polyhedron import Polyhedron
        from sympy.abc import w, x, y, z
        pgroup = [Permutation([[0, 1, 2], [3]]),
                  Permutation([[0, 1, 3], [2]]),
                  Permutation([[0, 2, 3], [1]]),
                  Permutation([[1, 2, 3], [0]]),
                  Permutation([[0, 1], [2, 3]]),
                  Permutation([[0, 2], [1, 3]]),
                  Permutation([[0, 3], [1, 2]]),
                  Permutation([[0, 1, 2, 3]])]
        corners = [w, x, y, z]
        faces = [(w, x, y), (w, y, z), (w, z, x), (x, y, z)]
>       assert _test_args(Polyhedron(corners, faces, pgroup))

sympy/core/tests/test_args.py:514: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
sympy/combinatorics/polyhedron.py:389: in __new__
    faces = [minlex(f, directed=False) for f in faces]
sympy/combinatorics/polyhedron.py:389: in <listcomp>
    faces = [minlex(f, directed=False) for f in faces]
sympy/utilities/iterables.py:2430: in minlex
    best = rotate_left(seq, least_rotation(seq))
sympy/utilities/iterables.py:1309: in least_rotation
    if (sj) < (S[k]):

self = x < w

    def __bool__(self):
>       raise TypeError("cannot determine truth value of Relational")
E       TypeError: cannot determine truth value of Relational

sympy/core/relational.py:395: TypeError
```

--- a/sympy/combinatorics/polyhedron.py
+++ b/sympy/combinatorics/polyhedron.py
@@ -3,7 +3,7 @@ from sympy.combinatorics.perm_groups import PermutationGroup
 from sympy.core import Basic, Tuple
 from sympy.core.compatibility import as_int
 from sympy.sets import FiniteSet
-from sympy.utilities.iterables import (minlex, unflatten, flatten)
+from sympy.utilities.iterables import (minlex, unflatten, flatten, default_sort_key)
 
 rmul = Perm.rmul
 
@@ -386,7 +386,7 @@ class Polyhedron(Basic):
         .. [1] www.ocf.berkeley.edu/~wwu/articles/platonicsolids.pdf
 
         """
-        faces = [minlex(f, directed=False) for f in faces]
+        faces = [minlex(f, directed=False, key=default_sort_key) for f in faces]
         corners, faces, pgroup = args = \
             [Tuple(*a) for a in (corners, faces, pgroup)]
         obj = Basic.__new__(cls, *args)

--- a/sympy/utilities/iterables.py
+++ b/sympy/utilities/iterables.py
@@ -9,9 +9,8 @@ from operator import gt
 from sympy.core import Basic
 
 # this is the logical location of these functions
-from sympy.core.compatibility import (
-    as_int, default_sort_key, is_sequence, iterable, ordered
-)
+from sympy.core.compatibility import (as_int, is_sequence, iterable, ordered)
+from sympy.core.compatibility import default_sort_key  # noqa: F401
 
 import sympy
 

--- a/sympy/utilities/iterables.py
+++ b/sympy/utilities/iterables.py
@@ -13,6 +13,8 @@ from sympy.core.compatibility import (
     as_int, default_sort_key, is_sequence, iterable, ordered
 )
 
+import sympy
+
 from sympy.utilities.enumerative import (
     multiset_partitions_taocp, list_visitor, MultisetPartitionTraverser)
 
@@ -1274,7 +1276,7 @@ def rotate_right(x, y):
     return x[y:] + x[:y]
 
 
-def least_rotation(x):
+def least_rotation(x, key=None):
     '''
     Returns the number of steps of left rotation required to
     obtain lexicographically minimal string/list/tuple, etc.
@@ -1295,6 +1297,7 @@ def least_rotation(x):
     .. [1] https://en.wikipedia.org/wiki/Lexicographically_minimal_string_rotation
 
     '''
+    if key is None: key = sympy.Id
     S = x + x      # Concatenate string to it self to avoid modular arithmetic
     f = [-1] * len(S)     # Failure function
     k = 0       # Least rotation of string found so far
@@ -1302,11 +1305,11 @@ def least_rotation(x):
         sj = S[j]
         i = f[j-k-1]
         while i != -1 and sj != S[k+i+1]:
-            if default_sort_key(sj) < default_sort_key(S[k+i+1]):
+            if key(sj) < key(S[k+i+1]):
                 k = j-i-1
             i = f[i]
         if sj != S[k+i+1]:
-            if default_sort_key(sj) < default_sort_key(S[k]):
+            if key(sj) < key(S[k]):
                 k = j
             f[j-k] = -1
         else:
@@ -2398,7 +2401,7 @@ def generate_oriented_forest(n):
                 break
 
 
-def minlex(seq, directed=True):
+def minlex(seq, directed=True, key=None):
     """
     Return the rotation of the sequence in which the lexically smallest
     elements appear first, e.g. `cba ->acb`.
@@ -2408,6 +2411,8 @@ def minlex(seq, directed=True):
 
     If ``directed`` is False then the smaller of the sequence and the
     reversed sequence is returned, e.g. `cba -> abc`.
+
+    If ``key`` is not None then it is used to extract a comparison key from each element in iterable.
 
     Examples
     ========
@@ -2425,13 +2430,19 @@ def minlex(seq, directed=True):
     >>> minlex('11010011000', directed=False)
     '00011001011'
 
+    >>> minlex(('bb', 'aaa', 'c', 'a'))
+    ('a', 'bb', 'aaa', 'c')
+    >>> minlex(('bb', 'aaa', 'c', 'a'), key=len)
+    ('c', 'a', 'bb', 'aaa')
+
     """
 
-    best = rotate_left(seq, least_rotation(seq))
+    if key is None: key = sympy.Id
+    best = rotate_left(seq, least_rotation(seq, key=key))
     if not directed:
         rseq = seq[::-1]
-        rbest = rotate_left(rseq, least_rotation(rseq))
-        best = min(best, rbest, key=default_sort_key)
+        rbest = rotate_left(rseq, least_rotation(rseq, key=key))
+        best = min(best, rbest, key=key)
 
     # Convert to tuple, unless we started with a string.
     return tuple(best) if not isinstance(seq, str) else best

--- a/sympy/utilities/tests/test_iterables.py
+++ b/sympy/utilities/tests/test_iterables.py
@@ -645,6 +645,7 @@ def test_minlex():
     assert minlex((1, 0, 2)) == (0, 2, 1)
     assert minlex((1, 0, 2), directed=False) == (0, 1, 2)
     assert minlex('aba') == 'aab'
+    assert minlex(('bb', 'aaa', 'c', 'a'), key=len) == ('c', 'a', 'bb', 'aaa')
 
 
 def test_ordered():

